dask/dataframe/io/parquet.py

```diff
@@ -1107,51 +1056,62 @@ def read_parquet(path, columns=None, filters=None, categories=None, index=None,
     --------
     to_parquet
     """
-    is_ParquetFile = False
-    try:
-        import fastparquet
-        if isinstance(path, fastparquet.api.ParquetFile):
-            if path.open != fastparquet.util.default_open:
-                assert (re.match('.*://', path.fn)), \
-                       ("ParquetFile: Path must contain protocol" +
-                        " (e.g., s3://...) when using other than the default" +
-                        " LocalFileSystem. Path given: " + path.fn)
-
-            assert (engine in ['auto', 'fastparquet']), \
-                   ("'engine' should be set to 'auto' or 'fastparquet' " +
-                    'when reading from fastparquet.ParquetFile')
-            is_ParquetFile = True
-    except ImportError:
-            pass
-
-    if is_ParquetFile:
-        read = get_engine('fastparquet')['read']
-        if path.fn.endswith('_metadata'):
-            # remove '_metadata' from path
-            urlpath = path.fn[:-len('_metadata')]
-        else:
-            urlpath = path.fn
+    if isinstance(columns, str):
+        df = read_parquet(path, [columns], filters, categories, index,
+                          storage_options, engine, infer_divisions)
+        return df[columns]
 
-        fs, fs_token, paths = get_fs_token_paths(
-            urlpath,
-            mode='rb',
-            storage_options=storage_options
-        )
-        paths = path
+    name = 'read-parquet-' + tokenize(path, columns, filters, categories,
+                                      index, storage_options, engine,
+                                      infer_divisions)
+
+    read = get_engine(engine)['read']
+    fs, fs_token, paths = get_fs_token_paths(
+        path, mode='rb',
+        storage_options=storage_options
+    )
+
+    paths = sorted(paths, key=natural_sort_key)  # numeric rather than glob ordering
+
+    func, meta, statistics, parts = read(fs, fs_token, paths, columns=columns, filters=filters,
```

Comment:
Do we actually use the fs token? Seems like `open_files` would be cleaner, if possible, and I see the name, above, doesn't use it.

Reply:
I don't know.